src/components/color_picker/color_picker.tsx

```diff
@@ -248,48 +324,91 @@ export const EuiColorPicker: FunctionComponent<EuiColorPickerProps> = ({
 
   const handleColorInput = (e: React.ChangeEvent<HTMLInputElement>) => {
     handleOnChange(e.target.value);
-    if (chromaValid(e.target.value)) {
-      updateColorAsHsv(chroma(e.target.value).hsv());
+    const newColor = getChromaColor(e.target.value, showAlpha);
+    if (newColor) {
+      updateColorAsHsv(newColor.hsv());
+    }
+  };
+
+  const updateWithHsv = (hsv: ColorSpaces['hsv']) => {
+    const color = chroma.hsv(...hsv).alpha(alphaChannel);
+    let formatted;
+    if (preferredFormat === 'rgba') {
+      formatted =
+        alphaChannel < 1
+          ? color.rgba().join(RGB_JOIN)
+          : color.rgb().join(RGB_JOIN);
+    } else {
+      formatted = color.hex();
     }
+    handleOnChange(formatted);
+    updateColorAsHsv(hsv);
   };
 
   const handleColorSelection = (color: ColorSpaces['hsv']) => {
-    const [h] = colorAsHsv;
+    const [h] = usableHsv;
     const [, s, v] = color;
     const newHsv: ColorSpaces['hsv'] = [h, s, v];
-    handleOnChange(chroma.hsv(...newHsv).hex());
-    updateColorAsHsv(newHsv);
+    updateWithHsv(newHsv);
   };
 
   const handleHueSelection = (hue: number) => {
-    const [, s, v] = colorAsHsv;
+    const [, s, v] = usableHsv;
     const newHsv: ColorSpaces['hsv'] = [hue, s, v];
-    handleOnChange(chroma.hsv(...newHsv).hex());
-    updateColorAsHsv(newHsv);
+    updateWithHsv(newHsv);
   };
 
   const handleSwatchSelection = (color: string) => {
+    const newColor = getChromaColor(color, showAlpha);
     handleOnChange(color);
-    updateColorAsHsv(chroma(color).hsv());
+    if (newColor) {
+      updateColorAsHsv(newColor.hsv());
+    }
 
     handleFinalSelection();
   };
 
+  const handleAlphaSelection = (
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.MouseEvent<HTMLButtonElement>,
+    isValid: boolean
+  ) => {
+    const target = e.target as HTMLInputElement;
+    setAlphaRangeValue(target.value || '');
+    if (isValid) {
+      const alpha = parseInt(target.value, 10) / 100;
+      const newColor = chromaColor ? chromaColor.alpha(alpha) : null;
+      const hex = newColor ? newColor.hex() : HEX_FALLBACK;
+      const rgba = newColor ? newColor.rgba() : RGB_FALLBACK;
+      let text;
+      if (preferredFormat === 'rgba') {
+        text =
+          alpha < 1
+            ? [...rgba].join(RGB_JOIN)
+            : [...rgba].splice(0, 3).join(RGB_JOIN);
```

Comment:
Doesn't need to create two new arrays, can `slice` without mutating the source.
```suggestion
            : rgba.slice(0, 3).join(RGB_JOIN);
```